Number_types/include/CGAL/Quotient.h

```diff
@@ -686,11 +707,234 @@ template < class NT > class Real_embeddable_traits_quotient_base< Quotient<NT> >
     class To_interval
       : public CGAL::cpp98::unary_function< Type, std::pair< double, double > > {
       public:
+
+        bool are_correct_bounds( const double l, const double u, const Type& x ) const {
+
+          if (
+            CGAL::abs(l) == std::numeric_limits<double>::infinity() ||
+            CGAL::abs(u) == std::numeric_limits<double>::infinity() ||
+            CGAL::abs(l) == 0.0 || CGAL::abs(u) == 0.0) {
+            return true;
+          }
+          CGAL_assertion(CGAL::abs(l) != std::numeric_limits<double>::infinity());
+          CGAL_assertion(CGAL::abs(u) != std::numeric_limits<double>::infinity());
+          CGAL_assertion(CGAL::abs(l) != 0.0);
+          CGAL_assertion(CGAL::abs(u) != 0.0);
+
+          const Type lb(l), ub(u);
+          CGAL_assertion(lb <= ub);
+          CGAL_assertion(lb <= x);
+          CGAL_assertion(ub >= x);
+          return lb <= ub && lb <= x && ub >= x;
+        }
+
+        std::pair< Interval_nt<>, int64_t > get_interval_exp( NT& x ) const {
+
+          CGAL_assertion(x >= 0);
+          int64_t d = 0;
+          double l = 0.0, u = 0.0;
+          const int64_t n = static_cast<int64_t>(msb(x));
+          const int64_t num_dbl_digits = std::numeric_limits<double>::digits - 1;
+
+          if (n > num_dbl_digits) {
+            d = n - num_dbl_digits;
+            x = x >> d;
+            const NT y = x + 1;
+            l = static_cast<double>(x);
+            u = static_cast<double>(y);
+          } else {
+            l = static_cast<double>(x);
+            u = l;
+          }
+          return std::make_pair( Interval_nt<>(l, u), d );
+        }
+
+        std::pair<double, double> get_interval_as_gmpzf( Type x ) const {
+
+          CGAL_assertion_code(const Type input = x);
+          double l = 0.0, u = 0.0;
+          if (x.num == 0) { // return [0.0, 0.0]
```

Comment:
In the code of [`rational_adaptor`](https://github.com/boostorg/multiprecision/blob/rational_adaptor_standalone/include/boost/multiprecision/rational_adaptor.hpp#L33)  they have constants `zero` and `one`.

Reply:
which are just 0 and 1 in type `signed char` or `int` depending on the type. I'd rather use CGAL's is_zero and centralize the choice of how to do that in one place, the specialization of Is_zero.

Reply:
I fixed it using `iz_zero`, `is_negative`, and `is_positive`.